platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection getAttributes(String username) throws URISyntaxException {
+        ProcessedClaimCollection claimCollection = new ProcessedClaimCollection();
+
+        LOGGER.debug("Sending AttributeQuery Request.");
+
+        Assertion assertion = null;
+        try {
+            assertion = createAttributeQueryClient(simpleSign, externalAttributeStoreUrl, issuer,
+                    destination).retrieveResponse(username);
+        } catch (AttributeQueryException ex) {
+            LOGGER.error("Error occurred in AttributeQueryClient, did not retrieve response.", ex);
+        }
+
+        if (assertion != null) {
+            createClaims(claimCollection, assertion);
+        }
+        return claimCollection;
+    }
+
+    /**
+     * Creates claims based on the retrieved attributes.
+     *
+     * @param claimsCollection The collection of claims.
+     * @param assertion        Assertion of the response.
+     * @return The collection of claims.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection createClaims(ProcessedClaimCollection claimsCollection,
+            Assertion assertion) throws URISyntaxException {
+
+        AttributeStatement attributeStatement = assertion.getAttributeStatements().get(0);
+        List<Attribute> attributeList = attributeStatement.getAttributes();
+
+        // If the claim is supported, check if it has an attribute mapping.
+        // If so, map the attribute (attribute value -> mapped attribute value)
+        // and create the claim, otherwise, create the claim using its original attribute value.
+        for (Attribute attribute : attributeList) {
+            for (String claimType : supportedClaims) {
+                if (claimType.equalsIgnoreCase(attribute.getName())) {
+                    String claimValue = attribute.getDOM().getTextContent();
+                    if (attributeMap.containsKey(attribute.getDOM().getTextContent())) {
+                        claimsCollection.add(createSingleValuedClaim(claimType,
+                                attributeMap.get(claimValue)));
+                    } else {
+                        claimsCollection.add(createSingleValuedClaim(claimType, claimValue));
+                    }
+                    break;
+                }
+            }
+        }
+        return claimsCollection;
+    }
+
+    /**
+     * Creates a claim with a single value.
+     *
+     * @param claimType  The claim type.
+     * @param claimValue The claim value.
+     * @return The claim.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaim createSingleValuedClaim(String claimType, String claimValue)
+            throws URISyntaxException {
+        ProcessedClaim claim = new ProcessedClaim();
+
+        claim.setClaimType(new URI(claimType));
+        List<Object> claimValues = new ArrayList<>();
```

Comment:
Should this be an immutable/unmodifiable collection? An `ImmutableList.of(claimValue)` would be cleaner and safer in that case.